wasmcloud-test-util/src/provider_test.rs

```diff
@@ -2,58 +2,163 @@
 //!
 //! simple test harness to load a capability provider and test it
 //!
+
+pub use log;
+
 use crate::testing::TestResult;
 use anyhow::anyhow;
 use async_trait::async_trait;
-use futures::future::BoxFuture;
+use futures::{future::BoxFuture, lock::Mutex};
 use nkeys::{KeyPair, KeyPairType};
-use serde::Serialize;
-use std::{fs, io::Write, ops::Deref, path::PathBuf, sync::Arc};
+use serde::{de, Deserialize, Serialize};
+use std::{
+    collections::{BTreeMap, HashMap},
+    fs,
+    io::Write,
+    ops::Deref,
+    path::PathBuf,
+    str::FromStr,
+    sync::Arc,
+    time::Duration,
+};
 use tokio::sync::OnceCell;
-use toml::value::Value as TomlValue;
 use wasmbus_rpc::{
     common::{Context, Message, SendOpts, Transport},
     core::{HealthCheckRequest, HealthCheckResponse, HostData, LinkDefinition, WasmCloudEntity},
     error::{RpcError, RpcResult},
     rpc_client::RpcClient,
 };
 
+pub type TestFunc = fn() -> BoxFuture<'static, RpcResult<()>>;
+
 pub type SimpleValueMap = std::collections::HashMap<String, String>;
 pub type TomlMap = std::collections::BTreeMap<String, toml::Value>;
 pub type JsonMap = serde_json::Map<String, serde_json::Value>;
 
-const DEFAULT_START_DELAY_SEC: u64 = 1;
-const DEFAULT_RPC_TIMEOUT_MILLIS: u64 = 2000;
+const DEFAULT_START_DELAY: Duration = Duration::from_secs(1);
+const DEFAULT_RPC_TIMEOUT: Duration = Duration::from_millis(2000);
```

Comment:
super tiny nit - is there a reason why one duration is created from seconds and the other from millis?

Reply:
That was just my attempt to minimize the diff for easier review, will amend the commit